### PR TITLE
Bar graph

### PR DESCRIPTION
**Pull request by kleekich**
@htlin2 Please give me feedback after code review! Thanks!

### CODE REVIEW

**Comment by kleekich** [comment]
@htlin2 Please give me feedback after code review! Thanks!

File: client/src/components/PriceVolumeChart.jsx [hunk 1]
```diff
@@ -13,18 +14,24 @@ class PriceVolumeChart extends React.Component {
...
+        this.findBarGraphIndex = this.findBarGraphIndex.bind(this);
+        this.indexInRange = this.indexInRange.bind(this);
+        this.barWidth = 11.457627118644067;
     }
+
     componentDidMount(){
         this.handleFetch();
     }
 
     handleFetch(id = '5bd277855f59f4e021dd7a62'){
```
Comment (htlin2) [inline client/src/components/PriceVolumeChart.jsx]: We should talk about the id's we are using as a group, as all ids that are generated in MongoDB are unique.

File: client/src/components/PriceVolumeChart.jsx [hunk 2]
```diff
@@ -34,25 +41,66 @@ class PriceVolumeChart extends React.Component {
...
+                            this.state.volumes.map((h, i)=>{
+                                return(
+                                <rect rx="1" x={this.barWidth*2*i} y="0" width={this.barWidth} height={h} fill={this.indexInRange(i) ? "#20ce99" : "#0e0d0d"}>
+                                </rect>
+                                )
+                            })
+                        }
+                    </g>
+                    <g transform="translate(0,200) scale(1,-1)">
+                        <line x1="0" x2="676" y1="10" y2="10" stroke="#8c8c8e"></line>
```
Comment (htlin2) [inline client/src/components/PriceVolumeChart.jsx]: can use react self closing tags for < line /> for cleaner code

File: client/src/components/PriceVolumeChart.jsx [hunk 3]
```diff
@@ -34,25 +41,66 @@ class PriceVolumeChart extends React.Component {
...
+                        <line x1={this.state.currentPriceIndex * this.barWidth * 2} x2={this.state.currentPriceIndex * this.barWidth * 2} y1="10" y2="130" stroke="#20ce99"></line>
```
Comment (htlin2) [inline client/src/components/PriceVolumeChart.jsx]: can use react self closing tags for < line /> for cleaner code

File: client/src/components/PriceVolumeChart.jsx [hunk 4]
```diff
@@ -34,25 +41,66 @@ class PriceVolumeChart extends React.Component {
...
+                        <circle r="5" cx={this.state.currentPriceIndex * this.barWidth * 2} cy="10" fill="#20ce99"></circle>
```
Comment (htlin2) [inline client/src/components/PriceVolumeChart.jsx]: can use react self closing tags for < circle /> for cleaner code

File: client/src/components/PriceVolumeChart.jsx [hunk 5]
```diff
@@ -34,25 +41,66 @@ class PriceVolumeChart extends React.Component {
...
             <div>
                 <h1>Price Paid on Robinhood</h1>
-                <h2>Name: <span id='name'>{this.state.name}</span></h2>
-                <h2>52 Week Low: $<span id='lowest'>{this.state.lowest}</span></h2>
-                <h2>52 Week High: $<span id='heighest'>{this.state.heighest}</span></h2>
-                <h2>Average Price Paid: $<span id='average'>{this.state.averagePrice.toFixed(2)}</span></h2>
-                <h2>Current Price: $<span id='current'>{this.state.currentPrice}</span></h2>
-                <h2 id='difference'><span>{Math.abs(this.state.difference)}</span>% {this.state.difference>0 ? 'Higher':'Lower'}</h2>
+                <svg viewBox="0 0 676 200" width="676" height="200">
+                    <rect x="0" y="0" width="676" height="200" id="bargraph" fill="#1b1b1d"></rect>
```
Comment (htlin2) [inline client/src/components/PriceVolumeChart.jsx]: can use react self closing tags for < rect /> for cleaner code

File: client/src/components/PriceVolumeChart.jsx [hunk 6]
```diff
@@ -34,25 +41,66 @@ class PriceVolumeChart extends React.Component {
...
+                    <g transform="translate(0,180) scale(1,-1)">
+                        {
+                            this.state.volumes.map((h, i)=>{
+                                return(
+                                <rect rx="1" x={this.barWidth*2*i} y="0" width={this.barWidth} height={h} fill={this.indexInRange(i) ? "#20ce99" : "#0e0d0d"}>
```
Comment (htlin2) [inline client/src/components/PriceVolumeChart.jsx]: can use react self closing tags for < rect /> for cleaner code and maybe indent after return ?

**Review by htlin2 — COMMENTED** [review]
Great job! Everything looks great. Maybe some self closing tags for cleaner code?

File: client/src/components/PriceVolumeChart.jsx (hunk 1 above)
Comment (kleekich) [reply client/src/components/PriceVolumeChart.jsx]: Yes. As we discussed earlier, I would refactor my data schema.

File: client/src/components/PriceVolumeChart.jsx (hunk 2 above)
Comment (kleekich) [reply client/src/components/PriceVolumeChart.jsx]: I applied to all related elements. Thanks!

File: client/src/components/PriceVolumeChart.jsx (hunk 3 above)
Comment (kleekich) [reply client/src/components/PriceVolumeChart.jsx]: Yes. I applied self-closing tags to all the elements if needed.

File: client/src/components/PriceVolumeChart.jsx (hunk 4 above)
Comment (kleekich) [reply client/src/components/PriceVolumeChart.jsx]: Yes. I applied self-closing tags to all the elements if needed.

File: client/src/components/PriceVolumeChart.jsx (hunk 5 above)
Comment (kleekich) [reply client/src/components/PriceVolumeChart.jsx]: Yes. I applied self-closing tags to all the elements if needed.

File: client/src/components/PriceVolumeChart.jsx (hunk 6 above)
Comment (kleekich) [reply client/src/components/PriceVolumeChart.jsx]: Yes. I applied self-closing tags to all the elements if needed.

**Comment by kleekich** [comment]
@htlin2 Can you confirm my changes? Thanks.

**Comment by htlin2** [comment]
Everything looks great!

**Comment by kleekich** [comment]
Thank you!

**Review by htlin2 — APPROVED** [review]
Everything Looks Great!!!!

**Pull request merged**